src/gravel/controllers/nodes/mgr.py

```diff
@@ -650,3 +654,49 @@ async def _handle_ready_to_add(
         if not mon.set_replicated_ruleset():
             logger.error(
                 "handle ready to add > unable to set replicated ruleset")
+
+        await self._set_pool_default_size()
+
+    async def _set_pool_default_size(self) -> None:
+        """ reset the osd pool default size """
+        assert self._init_stage == NodeInitStage.STARTED
+
+        def get_target_size():
+            orch: Orchestrator = Orchestrator(self.gstate.ceph_mgr)
+            orch_hosts: List[OrchHostListModel] = orch.host_ls()
+            return 2 if len(orch_hosts) < 3 else 3
+
+        mon: Mon = self.gstate.ceph_mon
+        current_size: int = mon.get_pool_default_size()
```

Comment:
This can technically be `None` in case of error, so it might be better to check it, or at least `assert` if so. Better yet would be to catch a raised exception.

Reply:
good catch, I've added handling similar to how we deal failures during `set_replicated_ruleset` et al.